src/libraries/System.Net.Http/src/System/Net/Http/Metrics/HttpMetricsEnrichmentContext.cs

```diff
@@ -0,0 +1,162 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics;
+
+namespace System.Net.Http.Metrics;
+
+public class HttpMetricsEnrichmentContext
+{
+    private HttpRequestMessage? _request;
+    private HttpResponseMessage? _response;
+    private Exception? _exception;
+    private TagCollection _tags;
```

Comment:
If we are pooling, I'd suggest a List is fine. While I'm not opposed if you want to keep the custom type with more checks around improper use, it felt like overkill to me. If we weren't pooling then I'd be more worried about putting guards in place to allow us to do it in the future, but if we are pooling it tends to be self-policing. Code that tries to reuse pooled objects usually quickly doesn't work.